### PR TITLE
revised SSS.6,7,8,9,12,13,14 based on the approved corresponding SCS.6,7,8,912,13,14 test cases

### PR DESCRIPTION
**Pull request by jni2000**
Federated Wireless revised SSS.6,7,8,9,10,12,13,14,15 based on the approved corresponding SCS.6,7,8,9,10,12,13,14,15 test cases. Please review and give your comments.

### CODE REVIEW

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    """
+    config = loadConfig(config_filename)
+    try:
+      self.assertTlsHandshakeFailure(client_cert=config['sasCert'],
+                                     client_key=config['sasKey'])
+    except AssertionError as e:
+      self.SasReset()
+      
+      # Initiate Full Activity Dump
+      self._sas_admin.TriggerFullActivityDump()
```
Comment (norbertbarbosa) [inline src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: sas_admin.TriggerFullActivityDump() and sas.GetFullActivityDump() don't currently exist in sas_interface.py and sas.py. As I understand, they are in review in #296.
So not sure if we need to wait for this PR before continuing this review, or if you want to 'merge' only the minimal required part: only the functions TriggerFullActivityDump() and GetFullActivityDump() from file sas_interface.py, sas.py and fake_sas.py

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      try:
+         self._sas.GetFullActivityDump()
```
Comment (norbertbarbosa) [inline src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: if missing code backported, you are still missing the `ssl_cert` and `ssl_key`parameters

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+    # Load the keys/certs and check that TLS handshake is valid
+    sas_cert = config['sasCert']
+    sas_key = config['sasKey']
+    self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'], sas_cert, sas_key)
+    
+    # Initiate Full Activity Dump
+    self._sas_admin.TriggerFullActivityDump() 
+    try:
+      self._sas.GetFullActivityDump()
```
Comment (norbertbarbosa) [inline src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: idem as SCS_10: missing `ssl_cert` and `ssl_key`parameters

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py (hunk 1 above)
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: we took SSS.10 and SSS.15 out to wait for the #296 approval.

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py (hunk 2 above)
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: The test cases SSS.10 and SSS.15 have been removed. These test cases have dependency on another PR#296.

File: src/harness/testcases/WINNF_FT_S_SSS_testcase.py (hunk 3 above)
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_SSS_testcase.py]: The test cases SSS.10 and SSS.15 have been removed. These test cases have dependency on another PR#296.

**Comment by jni2000** [comment]
Norbert

We have taken the following actions to address your review comments:
1. added in README the missing the newly added certificates
2. took SSS.10 and SSS.15 out to wait for #296 approval.

Please go ahead review this submission and approve if no other concerns.

Thanks

File: src/harness/certs/README.md [hunk 4]
```diff
@@ -99,3 +104,25 @@ Required certificates are:
...
   Used on security test test_WINNF_FT_S_SCS_15.
+
+* `unrecognized_sas.[cert|key]`: leaf SAS certificate signed by
+  `unrecognized_root_ca`, and corresponding trusted client certificates bundle.
+  Used on security test test_WINNF_FT_S_SSS_6.
+  
+* `corrupted_sas.cert`: corrupted 'sas.cert' certificate where the 20th character have been changed.
+  Used on security test test_WINNF_FT_S_SSS_7.
+  
+* `self_signed_client.cert`: self signed certificate of SAS signed by sas.key
```
Comment (norbertbarbosa) [inline src/harness/certs/README.md]: should be `self_signed_sas.cert` ?

Comment (jni2000) [reply src/harness/certs/README.md]: thanks, fixed.

**Comment by jni2000** [comment]
Thanks, Norbert, for your time in reviewing these test cases. Really appreciate your help on reaching out end of February delivery goal.

**Comment by kate-harrison** [comment]
James, it seems that the SAS Test Harness is not injected in these test cases. I thought we had discussed over email and agreed to add that in?

**Comment by jni2000** [comment]
You are right. I will take it back and do a quick patch as soon as possible.

James Ni
Director of SAS System
www.federatedwireless.com<http://www.federatedwireless.com/>
https://www.linkedin.com/pub/james-ni/4/3b9/24a
978.996.7480
[Description: Macintosh HD:Users:Sam:Dropbox (Bluetext):Federated Wireless:Creative:Logo:Logo Lock Up:Final-Lockup:Federated Wireless No Tag:FedWireless-lockup-notag-web.png]

**Comment by kate-harrison** [comment]
Thanks! It's OK if you want to just do a followup PR -- no need to undo the changes in this PR first. I think that will be the fastest for everyone.

**Comment by jni2000** [comment]
Ok, will do. Thanks.

James Ni
Director of SAS System
www.federatedwireless.com<http://www.federatedwireless.com/>
https://www.linkedin.com/pub/james-ni/4/3b9/24a
978.996.7480
[Description: Macintosh HD:Users:Sam:Dropbox (Bluetext):Federated Wireless:Creative:Logo:Logo Lock Up:Final-Lockup:Federated Wireless No Tag:FedWireless-lockup-notag-web.png]

**Pull request merged**